packages/go_router_builder/lib/src/route_config.dart

```diff
@@ -63,6 +63,58 @@ extension $_extensionName on $_className {
   String get routeDataClassName => 'ShellRouteData';
 }
 
+/// The configuration to generate class declarations for a StatefulShellRouteData.
+class StatefulShellRouteConfig extends RouteBaseConfig {
+  StatefulShellRouteConfig._({
+    required this.navigatorKey,
+    required super.routeDataClass,
+    required super.parent,
+    required super.parentNavigatorKey,
+  }) : super._();
+
+  /// The command for calling the navigator key getter from the ShellRouteData.
+  final String? navigatorKey;
+
+  @override
+  Iterable<String> classDeclarations() => <String>[
+        '''
+extension $_extensionName on $_className {
+  static $_className _fromState(GoRouterState state) => const $_className();
+}
+'''
+      ];
+
+  @override
+  String get routeConstructorParameters =>
+      navigatorKey == null ? '' : 'navigatorKey: $navigatorKey,';
```

Comment:
no navigator key, but we need to have a way to pass in
`navigatorContainerBuilder` and `restorationScopeId`

Reply:
done